### PR TITLE
Add support for Amasty Shiprestriction

### PR DESCRIPTION
**Pull request by dumega**
# Description
For each rule created by the plugin Amasty Shiprestriction, there is a field to determine whether it can be applied in the admin area, and in the code it depends on the conditional statement `$this->appState->getAreaCode() == \Magento\Backend\App\Area\FrontNameResolver::AREA_CODE;`.

But for the Bolt pre-auth create order webhook, $this->appState->getAreaCode() returns \Magento\Framework\App\Area::AREA_WEBAPI_REST, therefore, an invalid rule can be applied to the cart by mistake.

Solution: Add filter('for_admin', 1) to rules collection when getting restriction rules for Bolt backoffice order via plugin methods.

Fixes: https://app.asana.com/0/1124368832381302/1202087335187579/f

#changelog Add support for Amasty Shiprestriction

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Review by mikeshutov-bolt — APPROVED** [review]

File: Plugin/Amasty/Shiprestriction/Model/ShippingRestrictionRulePlugin.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
```
Comment (johnmccombs1) [inline Plugin/Amasty/Shiprestriction/Model/ShippingRestrictionRulePlugin.php:14]: 2022

File: Plugin/Amasty/CommonRules/Model/ResourceModel/Rule/CollectionPlugin.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
```
Comment (johnmccombs1) [inline Plugin/Amasty/CommonRules/Model/ResourceModel/Rule/CollectionPlugin.php:14]: Copyright should be 2022

**Review by johnmccombs1 — APPROVED** [review]
LGTM

File: Plugin/Amasty/Shiprestriction/Model/ShippingRestrictionRulePlugin.php (hunk 1 above)
Comment (dumega) [reply Plugin/Amasty/Shiprestriction/Model/ShippingRestrictionRulePlugin.php]: Fixed

File: Plugin/Amasty/CommonRules/Model/ResourceModel/Rule/CollectionPlugin.php (hunk 2 above)
Comment (dumega) [reply Plugin/Amasty/CommonRules/Model/ResourceModel/Rule/CollectionPlugin.php]: Fixed

**Pull request merged**